sim/midas/src/main/cc/simif.h

```diff
@@ -41,12 +41,30 @@ class simif_t
     CLOCKBRIDGEMODULE_struct * clock_bridge_mmio_addrs;
     midas_time_t sim_start_time;
 
+    std::string blockingFail = "The test environment has starved the simulator, preventing forward progress.";
+
     inline void take_steps(size_t n, bool blocking) {
       write(this->master_mmio_addrs->STEP, n);
       if (blocking) while(!done());
     }
     virtual void load_mem(std::string filename);
 
+    bool waitOn(size_t flag_addr, double timeout) {
```

Comment:
ultra nit: maybe snake case the function names.

Reply:
Good call, gotta be C++-ing now :)